lib/mongo/grid_fs/download.ex

```diff
@@ -0,0 +1,80 @@
+defmodule Mongo.GridFs.Download do
+  @moduledoc """
+  The main entry point for downloading files from the grid-fs specified by the bucket struct.
+  """
+
+  alias BSON.ObjectId
+  alias Mongo.GridFs.Bucket
+
+  @type result :: {:error, :unknown} | {:error, :length_is_zero} | {:error, :not_found} | {:ok, Mongo.cursor}
+
+  @doc """
+  Opens a Stream from which the application can read the contents of the stored file
+  specified by fileId.
+  Returns a Stream.
+  """
+  @spec open_download_stream(Bucket.t, String.t | BSON.ObjectId.t | map()) :: result
+  def open_download_stream(bucket, file_id) when is_binary(file_id) do
+    find_one_file(bucket, %{"_id" => ObjectId.decode!(file_id)})
+  end
+
+  @doc """
+  Same as above, accepting an OID
```

Comment:
Don't say "Same as above". This is all the same function. Document it as a single function.